lib/fathead/pika/parse.py

```diff
@@ -0,0 +1,296 @@
+# -*- coding: utf-8 -*-
+
+from bs4 import BeautifulSoup
+import re
+import logging
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger()
+
+URL_ROOT = "https://pika.readthedocs.io/en/latest/"
+MODULES_URL_SUFFIX = {
+    'BlockingConnection' : 'modules/adapters/blocking.html',
+    'Select Connection Adapter' : 'modules/adapters/select.html',
+    'Tornado Connection Adapter' : 'modules/adapters/tornado.html',
+    'Twisted Connection Adapter' : 'modules/adapters/twisted.html',
+    'Channel' : 'modules/channel.html',
+    'Connection' : 'modules/connection.html',
+    'Authentication Credentials' : 'modules/credentials.html',
+    'Exceptions' : 'modules/exceptions.html',
+    'Connection Parameters' : 'modules/parameters.html',
+    'pika.spec' : 'spec.html'
+}
+
+
+def replace_all(text, terms):
+    """
+    Replace all the terms contained
+    in a dict {'from':'to'}.
+    """
+    for _from, _to in terms.items():
+        text = text.replace(_from, _to)
+
+    return text
+
+
+def format_signature(signature):
+    """
+    Remove unwanted characters from the
+    signature of various classes and methods.
+    """
+    terms = {
+        '\n' : '',
+        '[source]' : '',
+        u'¶' : ''
+    }
+
+    return replace_all(signature, terms)
+
+
+def build_abstract(info, example=None, detailed_info={}):
+    """
+    Constructs the abstract of an entry according to
+    the new fathead HTML template elements.
+    """
+    abstract = ''
+    if info:
+        abstract += '<p>%s</p' % info.replace('\n', '\\n')
```

Comment:
@sriranganathan `abstract += '<p>%s</p' % info.replace('\n', '\\n')`

The `<p>` is not closing correctly!